libs/gotrue/src/api.rs

```diff
@@ -11,15 +11,16 @@ use gotrue_entity::dto::{
 use gotrue_entity::error::{GoTrueError, GoTrueErrorSerde, GotrueClientError};
 use gotrue_entity::sso::{SSOProvider, SSOProviders};
 use infra::reqwest::{check_response, from_body, from_response};
+use reqwest_middleware::ClientWithMiddleware;
 
 #[derive(Clone)]
 pub struct Client {
-  client: reqwest::Client,
+  client: ClientWithMiddleware,
```

Comment:
How about we design a custom struct that encapsulates the ClientWithMiddleware? This approach would simplify switching to a different client in the future. Maybe it's easier to do more optimization without changing the interface.

Reply:
make sense